### PR TITLE
scripts: Organize dev-only scripts and use a single scripts/tls location

### LINKED ISSUE #518: scripts: Confusing/conflicting paths for cert-gen script
ItThe [documentation](https://github.com/coreos/matchbox/blame/d4e33efb38b326dafe961ba03c1936f47f36f26a/Documentation/deployment.md#L138) for matchbox refers to the script `cert-gen` used to generate TLS assets.  When searching through the release tarballs, this script seems to be in the location:

```
core@matchbox ~/matchbox-v0.6.0-linux-amd64 $ find * -name cert-gen
scripts/tls/cert-gen
```

While searching through the source repository this script is located at:

```
$ find * -name cert-gen
examples/etc/matchbox/cert-gen
```

This is due to the fact that in the [`release-files`](https://github.com/coreos/matchbox/blob/d4e33efb38b326dafe961ba03c1936f47f36f26a/scripts/release-files#L22) script the location of the path is changed.  It would be ideal if one of the following could occur:

  - standardize the path of where this script lives
  - change the documentation to make it clear if it is referencing a release asset or a source checkout

### PR DESCRIPTION
**Pull request by dghubble**
* Organize scripts which are only used for development under `scripts/dev` (not released)
* Move `examples/etc/matchbox/cert-gen` related files under `scripts/tls`. This matches the location within the release tarball. All docs can be updated to refer to `scripts/tls`, which is correct whether people clone the repo or download a tarball.
* Spruce up the TLS generation docs in general. Continue suggesting users prefer their own PKI, but acknowledge that many people use these self-signed certs and they're better than hand rolling your own (remove some scary messages).

This does not require a v0.6.1 release, just a docs sync. All prior official release tarballs are using the correct structure, we've opted to change to source code location instead.

cc @alekssaul 
Related ##521
Closes #518 
Closes https://github.com/coreos/tectonic-installer/issues/672

### CODE REVIEW

File: scripts/tls/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+## gRPC TLS Generation
+
+The Matchbox gRPC API allows clients (`terraform-provider-matchbox`) to create and update Matchbox resources. TLS credentials are used for client authentication and to establish a secure communication channel. When the gRPC API is [enabled](../../Documentation/deployment.md#customization), the server requires a TLS server certificate, key, and CA certificate ([locations](../../Documentation/config.md#files-and-directories)).
+
+The `cert-gen` helper script generates a self-signed CA, server certificate, and client certificate. **Prefer your organizations' PKI, if possible**
```
Comment (zbwright) [inline scripts/tls/README.md]: organization's

File: scripts/tls/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+Generate a `ca.crt`, `server.crt`, `server.key`, `client.crt`, and `client.key`.
+
+```sh
+$ ./cert-gen
+Creating FAKE CA, server cert/key, and client cert/key...
+...
+...
+...
+******************************************************************
+WARNING: Generated TLS credentials are ONLY SUITABLE FOR EXAMPLES!
```
Comment (zbwright) [inline scripts/tls/README.md]: SUITABLE ONLY FOR EXAMPLES

File: Documentation/deployment.md [hunk 3]
```diff
@@ -127,31 +127,39 @@ $ sudo firewall-cmd --zone=MYZONE --add-port=8080/tcp --permanent
...
 ```
 
-## Generate TLS credentials
+## Generate TLS Certificates
 
-*Skip this unless you need to enable the gRPC API*
+The Matchbox gRPC API allows clients (terraform-provider-matchbox) to create and update Matchbox resources. TLS credentials are needed for client authentication and to establish a secure communication channel. Client machines (those PXE booting) read from the HTTP endpoints and do not require this setup.
 
-The `matchbox` gRPC API allows client apps (terraform-provider-matchbox, Tectonic Installer, etc.) to update how machines are provisioned. TLS credentials are needed for client authentication and to establish a secure communication channel. Client machines (those PXE booting) read from the HTTP endpoints and do not require this setup.
+The `cert-gen` helper script generates a self-signed CA, server certificate, and client certificate. **Prefer your organizations' PKI, if possible**
```
Comment (zbwright) [inline Documentation/deployment.md]: organization's

File: scripts/tls/README.md (hunk 2 above)
Comment (dghubble) [inline scripts/tls/README.md]: I've relaxed this warning to be more allowing and just dropped the suitable only for examples text.

Comment (dghubble) [inline scripts/tls/README.md]: Its now:

```
WARNING: Generated credentials are self-signed. Prefer your
organization's PKI for production deployments.
```

**Comment by zbwright** [comment]
LGTM

**Comment by zbwright** [comment]
LGTM

**Pull request merged**